src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AsType.cs

```diff
@@ -63,17 +64,9 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 return false;
             }
 
-            // Check if table arg referrs to a connected data source.
-            var tableArg = args[1];
-            if (!binding.TryGetFirstNameInfo(tableArg.Id, out var tableInfo) ||
-                tableInfo.Data is not IExternalDataSource tableDsInfo ||
-                !(tableDsInfo is IExternalTabularDataSource))
-            {
-                errors.EnsureError(tableArg, TexlStrings.ErrAsTypeAndIsTypeExpectConnectedDataSource);
-                return false;
-            }
+            var tableDsInfo = argTypes[0].AssociatedDataSources.Single();
```

Comment:
>argTypes[0]. [](http://example.com/codeflow?start=30&length=12)

the tableDSInfo is from argtypes[1], see the removed code.

Reply:
Fixed